activerecord/lib/active_record/type/internal/timezone.rb

```diff
@@ -16,6 +16,10 @@ def is_utc?
         def default_timezone
           @timezone || ActiveRecord.default_timezone
         end
+
+        def ==(other)
+          super(other) && is_utc? == other.is_utc?
```

Comment:
Sounds like what should be compared here is the `@timezone` property, no?

Reply:
Ah yes, you are right. It's the same but it will be better. (The timezone can only have 2 acceptable values, :local or :utc). I'll change it.

Not sure it there is any concern regarding the note I added in the PR and whether this is a bandaid and we should merge it or the fix should be at the implementation level.